### DISCUSSION TITLE
Trouble connecting on cellular data

### DISCUSSION THREAD

**Discussion by BigRedProductions** [pivpn/pivpn#1615] [Support]
I am having trouble connecting to my VPN through cellular data. I can connect to it using WiFi, the same network the VPN is set up on. When I disconnect from WiFi, it won’t connect anymore until I connect back to WiFi. I have port forwarding set up. 

iPhone OpenVPN

**Comment by liasom** [comment]
Correction: Unlike BigRedProductions Using _cellular data I was able to successfully connect_ to the OpenVPN server yet was unable to send/receive any data and below is what worked for me to solve that issue. YMMV.

I has this issue with Android on the T-Mobile cellular network. OpenVPN running tcp over udp which as far as I know is the standard type of installation with PiVPN. Solved it after much research and by adding an "mssfix max" option to my client ovpn profile. There were T-Mobile mssfix values all over the place from multiple sources. So for me the value of mssfix max was determined by trial and error. Some speeds in the range of 1300 up to 1420 in increments of 10 (e'g 1300, 1310, 1320...) would work and others wouldn't. For me 1350 was the sweet spot that connected every time and best speeds measured by using OOKLA Speedtest. I know none of that is proper methodology but it got the problem solved. Here's the first few lines of the client.ovpn file after editing.

client
dev tun
proto udp
_**mssfix 1350**_
remote ......

When mssfix max is set, TCP sessions over UDP running on OpenVPN are made to limit their send packet sizes such that after OpenVPN has encapsulated TCP, the resulting UDP packet size will not exceed a maximum of the specified number of bytes. E.g. “mssfix 1400” will limit the UDP packet size to a maximum of 1400 bytes. Apparently the non-functional networking I encountered is because Cellular networks have a Maximum Transmission Unit (MTU) that’s lower than the OpenVPN default MTU (1500?). On the Cellular network if the UDP packets are larger than the MTU allows through the network they become fragmented. That’s the gist.

My client ovpn using the mssfix option still functions well over all WiFi I've used.

**Comment by coolapso** [comment]
Converting to Discussion > Support

**Comment by BigRedProductions** [comment]
I just got done trying this. It didn’t work for me.

**Comment by TinCanTech** [comment]
@BigRedProductions your problem of _not being able to connect over cellular network_ has absolutely **nothing** to do with an MSS value.

You may have more luck if you can determine the MTU of your cellular carrier.

**Reply by liasom** [reply]
Oops. Sorry I missed the detail that this was about _not being able to connect_. I could connect to the OpenVPN server using cellular data with tun0 MTU of 1500, but could not send/receive any data. I've edited my initial response above to indicate my error.

**Comment by BigRedProductions** [comment]
I have Verizon and what I was able to find is it’s MTU size is 1428, according to my research.

**Reply by TinCanTech** [reply]
Try adding `link-mtu 1380` to both your server and client config files.

It is not an ideal solution but I would be interested to know your results.

**Comment by BigRedProductions** [comment]
That didn’t work either. I’ve never had this much trouble before. This isn’t my first install of PiVPN.

**Reply by TinCanTech** [reply]
If you like, you can start a thread on the [Openvpn Forum](https://forums.openvpn.net/viewforum.php?f=6) and upload your complete server log at `--verb 4` for me to take a look. Otherwise, you will need to share your Server log on a paste site. eg: https://www.pastesite.org/

**Comment by TheRoarkster** [comment]
I had the same problem (on T-Mobile and, before that, the original Sprint network).  Adding the following to `/etc/openvpn/server.conf` solved it for me:
```
## Add to make it work with android
tun-mtu 1400
mssfix 1360
```

Probably an edge case, but this has happened to me repeatedly, so does it make any sense to add this to the script -- or an option for android?  Do these additions do any harm?
